docs/index.md

```diff
@@ -0,0 +1,388 @@
+
+### Closure Imports
+By default, the compiler generates code with Closure-style imports. If you
+specify a `library` option when running the compiler, the compiler creates a
+single `.js` file with your specified library name. Otherwise the compiler
+generates a `.js` file for each *message* in your `.proto` file. The names of
+the output files are computed by taking the `library` value or message name
+(lowercased), with the following changes.
+
+So, for example, let's say you invoke the compiler as follows:
+
+```
+protoc --proto_path=src --js_out=library=whizz/ponycopter,binary:build/gen src/foo.proto src/bar/baz.proto
+```
+
+The compiler will read the files `src/foo.proto` and `src/bar/baz.proto` and
+produce a single output file, `build/gen/whizz/ponycopter.js`. The compiler will
+automatically create the directory `build/gen/whizz` if necessary, but it will
+*not* create `build` or `build/gen`; they must already exist.
+The generated file(s) `goog.provide()` all 'the types defined in your `.proto`
+file(s), and `goog.require()` many types in the core protocol buffers library
+and Google Closure library. Make sure that your `goog.provide()` /
+`goog.require()` setup can find all of your generated code,
+[the core library `.js` files](https://github.com/protocolbuffers/protobuf-javascript)
+and the Google Closure library itself.
+
+You should be able to import your generated types with statements like:
+
+```js
+goog.require('proto.my.package.MyMessage');
+var message = proto.my.package.MyMessage();
+```
+
+### CommonJS Imports
+To specify that you want to use CommonJS-style imports instead of the default
+Closure style, you run the compiler with the `import_style=commonjs` option. The
+names of the output files are computed by taking the name of the each input
+`.proto` file and making two changes.
+
+**Note:** Specifying a `library` option is ignored with this import style.
+
+So, for example, let's say you invoke the compiler as follows:
+
+```
+protoc --proto_path=src --js_out=import_style=commonjs,binary:build/gen src/foo.proto src/bar/baz.proto
+```
+
+The compiler will read the files `src/foo.proto` and `src/bar/baz.proto` and
+produce two output files: `build/gen/foo_pb.js` and `build/gen/bar/baz_pb.js`.
+The compiler will automatically create the directory `build/gen/bar` if
+necessary, but it will *not* create `build` or `build/gen`; they must already
+exist.
+
+The generated code depends on the core runtime, which should be in a file
+called `google-protobuf.js`. If you installed protoc via `npm`, this file should
+already be built and available. If you are running from GitHub, you need to
+build it first by running:
+
+```
+PROTOC=/path/to/protoc PROTOC_INC=/path/to/proto/include gulp dist
+```
+
+You should be able to import your generated types with statements like:
+
+```js
+var messages = require('./messages_pb');
```

Comment:
nit: lets update these vars to const like its 2015

Reply:
done